### DISCUSSION TITLE
Why doesn't hy recognize packages in /.venv?

### DISCUSSION THREAD

**Discussion by Kyure-A** [hylang/hy#2483] [Q&A] [answered]
I am a newbie who has only known hylang for a few days. I use [Rye](https://github.com/mitsuhiko/rye) as my package manager. I don't know if this is a user-specific problem with Rye or a .venv problem, but .venv is not being read by hy (see log below).

```
> python3   
 Cannot read termcap database;
 using dumb terminal settings.
 Python 3.11.3 (main, May  7 2023, 19:29:55) [Clang 16.0.3 ] on linux
 Type "help", "copyright", "credits" or "license" for more information.
 >>> pprint.pprint(sys.path)
 Traceback (most recent call last):
   File "<stdin>", line 1, in <module>
 NameError: name 'pprint' is not defined. Did you mean: 'print'?
 >>> import pprint
 >>> pprint.pprint(sys.path)
 Traceback (most recent call last):
   File "<stdin>", line 1, in <module>
 NameError: name 'sys' is not defined
 >>> import sys
 >>> pprint.pprint(sys.path)
 ['',
  '/home/kyre/.rye/py/cpython@3.11.3/install/lib/python311.zip',
  '/home/kyre/.rye/py/cpython@3.11.3/install/lib/python3.11',
  '/home/kyre/.rye/py/cpython@3.11.3/install/lib/python3.11/lib-dynload',
  '/home/kyre/documents/{project_name}/.venv/lib/python3.11/site-packages',
  '/home/kyre/documents/{project_name}']
 >>> quit()
```
```
 > hy                 
 Cannot read termcap database;
 using dumb terminal settings.
 Hy 0.27.0 using CPython(main) 3.11.3 on Linux
 => (import sys pprint)    
 => (pprint.pprint sys.path)
 ['',
  '/home/kyre/.rye/tools/hy/bin',
  '/home/kyre/.rye/py/cpython@3.11.3/install/lib/python311.zip',
  '/home/kyre/.rye/py/cpython@3.11.3/install/lib/python3.11',
  '/home/kyre/.rye/py/cpython@3.11.3/install/lib/python3.11/lib-dynload',
  '/home/kyre/.rye/tools/hy/lib/python3.11/site-packages']
 => (quit)
```
The Python [Selenium Tutorial](https://www.selenium.dev/ja/documentation/webdriver/getting_started/first_script/) rewritten in Hylang does not work with the following error because the .venv is not loaded as logged.

```
  File "/home/kyre/.rye/shims/hy", line 8, in <module>
    sys.exit(hy_main())
             ^^^^^^^^^
  File "<frozen runpy>", line 291, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/home/kyre/documents/nitoku/src/classroom.hy", line 5, in <module>
    (import selenium [webdriver])
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'selenium'
```

What action should I take?

I am very aware that this is not the place to troubleshoot Rye, so if you think there is a problem with Rye, I would appreciate it if you would tell me so. Thanks.

**Comment by Kodiologist** [comment, marked as answer]
Hy is supposed to work with virtual environments; I use them all the time. So this could be a problem with Rye (which I don't know anything about), or a problematic interaction between Hy and Rye. It's also possible that you're running the wrong installation of Hy (i.e., not the one you installed in your virtual environment), which is a mistake I've made before. The message "Cannot read termcap database" suggests your environment is pretty screwed up.

**Reply by Kyure-A** [reply]
As pointed out, we were running hy installed in global, not hy in the virtual environment.
````
source .venv/bin/activate
````
I forgot the this, which solved the problem. Thanks a lot!
